File: Code_Forces/675-A.py
a, b, c = list(map(int, raw_input().split()))

if c == 0:
    if a == b:
        print 'YES'
    else:
        print 'NO'
else:
    if a == b:
        print 'YES'
    else:
        if a > b:
            if c > 0:
                print 'NO'
            else:
                if (a-b) % c == 0:
                    print 'YES'
                else:
                    print 'NO'
        elif a < b:
            if c < 0:
                print 'NO'
            else:
                if (a-b)%c == 0:
                    print 'YES'
                else:
                    print 'NO'

File: Code_Forces/675-A.c
#include<stdio.h>

int main(){
	
	int a, b ,c;
	scanf("%d %d %d",&a,&b,&c);
	if (a == b){
		printf("YES");
	}else{
		if (a > b){
			if (c > 0){
				printf("NO");
				}else{
					if ((a-b) % c == 0){
						printf("YES");
						}
					else{
						printf("NO");
						}
					}
			}else if(a < b){
				if (c < 0){
					printf("NO");
					}else{
						if ((a-b) % c == 0){
							printf("YES");
							}else{
								printf("NO");
								}
						}
				}
		}
	return 0;
}
